### DISCUSSION TITLE
Solidity Smart Contract Development : Remix doesnt compile codes

### DISCUSSION THREAD

**Discussion by GodwinDA** [Cyfrin/foundry-full-course-cu#2089] [General]
am just starting out on smart contract dev and been trying to run this code but it doesn't work. it just keeps compiling and stops without any result

// SPDX-License-Identifier: MIT
pragma solidity 0.8.18;

contract simplestorage { 

}

**Comment by CadetTalha** [comment]
the given line of codes is ok
if there is green tick on compiling the code then you are doing great and everything is fine 
for now that green tick will ensure that the code is ok but this code will not show you any output yet

**Comment by JoshuyPan** [comment]
Hi @GodwinDA, try using 0.8.19 or just add the carrot '^' in front of your solidity version as below. Save and compile.
Let me know
```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

contract simplestorage {

}
```

**Reply by GodwinDA** [reply]
still doesnt work, i dont know what to do as I cant move forward

tried other IDES but not good

**Reply by JoshuyPan** [reply]
try to run:
```bash
foundryup
```
into your terminal, check if you get some errors, and try again to compile.

**Reply by GodwinDA** [reply]
same issues bro

**Reply by JoshuyPan** [reply]
drop a screenshot please

**Reply by GodwinDA** [reply]
Works now, now on the latest version 0.8.26 
Thanks for the help bro

**Reply by JoshuyPan** [reply]
pleasure buddy, remember to close the discussion below, and never give up

**Comment by emu-code** [comment]
> am just starting out on smart contract dev and been trying to run this code but it doesn't work. it just keeps compiling and stops without any result
> 
> // SPDX-License-Identifier: MIT pragma solidity 0.8.18;
> 
> contract simplestorage {
> 
> }

Try changing your contract name to camelCase style like "SimpleStorage"

**Reply by GodwinDA** [reply]
not working

**Comment by EngrPips** [comment]
> am just starting out on smart contract dev and been trying to run this code but it doesn't work. it just keeps compiling and stops without any result
> 
> // SPDX-License-Identifier: MIT pragma solidity 0.8.18;
> 
> contract simplestorage {
> 
> }

If you are running in remix this issue might be related to your internet connection. you should check if your internet connection is stable.

**Reply by GodwinDA** [reply]
its not an internet connection issue

**Reply by EngrPips** [reply]
@GodwinDA did you figure out what the issue was?
